integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/DatabaseTest.java

```diff
@@ -0,0 +1,181 @@
+import com.sequenceiq.it.cloudbreak.newway.entity.database.DatabaseTestConnection;
+import com.sequenceiq.it.cloudbreak.newway.entity.database.DatabaseTestEntity;
+import com.sequenceiq.it.cloudbreak.newway.assertion.DatabaseExistsAssertion;
+import com.sequenceiq.it.cloudbreak.newway.assertion.DatabaseTestAccessDeniedAssertion;
+import com.sequenceiq.it.cloudbreak.newway.context.TestContext;
+import com.sequenceiq.it.util.LongStringGeneratorUtil;
+
+public class DatabaseTest extends AbstractIntegrationTest {
+
+    private static final String TEST_CONTEXT = "testContext";
+
+    private static final String DB_TYPE_PROVIDER = "databaseTypeTestProvider";
+
+    private static final String INVALID_ATTRIBUTE_PROVIDER = "databaseInvalidAttirbutesTestProvider";
+
+    private static final String DATABASE_PROTOCOL = "jdbc:postgresql://";
+
+    private static final String DATABASE_HOST_PORT_DB = "somedb.com:5432/mydb";
+
+    private static final String DATABASE_USERNAME = "username";
+
+    private static final String DATABASE_PASSWORD = "password";
+
+    private static final String BAD_REQUEST_KEY = "badRequest";
+
+    @Inject
+    private LongStringGeneratorUtil longStringGeneratorUtil;
+
+    @BeforeMethod
+    public void beforeMethod(Object[] data) {
+        TestContext testContext = (TestContext) data[0];
+        minimalSetupForClusterCreation(testContext);
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tear(Object[] data) {
+        ((TestContext) data[0]).cleanupTestContextEntity();
+    }
+
+    @Test(dataProvider = TEST_CONTEXT)
+    public void testCreateDeleteCreate(TestContext testContext) {
+        String databaseName = getNameGenerator().getRandomNameForMock();
+        testContext
+                .given(DatabaseEntity.class)
+                    .valid()
+                    .withName(databaseName)
+                .when(Database.post())
+                .when(Database.list())
+                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1))
+                .when(Database.deleteV2())
+                .when(Database.list())
+                .then(DatabaseExistsAssertion.getAssertion(databaseName, 0))
+                .when(Database.post())
+                .when(Database.list())
+                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1))
+                .validate();
+    }
+
+    @Test(dataProvider = TEST_CONTEXT)
+    public void testCreateTwice(TestContext testContext) {
+        String databaseName = getNameGenerator().getRandomNameForMock();
+        testContext
+                .given(DatabaseEntity.class)
+                    .valid()
+                    .withName(databaseName)
+                .when(Database.post())
+                .when(Database.list())
+                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1))
+                .when(Database.post(), key(BAD_REQUEST_KEY))
+                .expect(BadRequestException.class, key(BAD_REQUEST_KEY))
```

Comment:
What is the expected error msg?

Reply:
i think it is enough for checking the exception, error messages should be checked more likely in unit test